--- src.orig/java/lang/System.java	2013-01-04 14:04:02.000000000 -0500
+++ src/java/lang/System.java	2018-05-12 20:20:42.000000000 -0400
@@ -75,7 +75,8 @@
      * corresponds to keyboard input or another input source specified by
      * the host environment or user.
      */
-    public final static InputStream in = null;
+    // Modified for PolyLLVM.
+    public final static InputStream in = new BufferedInputStream(new FileInputStream(FileDescriptor.in));

     /**
      * The "standard" output stream. This stream is already
@@ -102,7 +103,8 @@
      * @see     java.io.PrintStream#println(java.lang.Object)
      * @see     java.io.PrintStream#println(java.lang.String)
      */
-    public final static PrintStream out = null;
+    // Modified for PolyLLVM.
+    public final static PrintStream out = new PrintStream(new BufferedOutputStream(new FileOutputStream(FileDescriptor.out), 128), true);

     /**
      * The "standard" error output stream. This stream is already
@@ -116,7 +118,8 @@
      * variable <code>out</code>, has been redirected to a file or other
      * destination that is typically not continuously monitored.
      */
-    public final static PrintStream err = null;
+    // Modified for PolyLLVM.
+    public final static PrintStream err = new PrintStream(new BufferedOutputStream(new FileOutputStream(FileDescriptor.err), 128), true);

     /* The security manager for the system.
      */
