--- src.orig/java/nio/charset/StandardCharsets.java	2018-10-11 16:26:48.859620877 -0400
+++ src/java/nio/charset/StandardCharsets.java	2018-10-11 16:32:21.167615891 -0400
@@ -1,3 +1,5 @@
+// This file's original contents have been modified
+// by a JLang patch on 2018-10-11 (the date this source code was compiled)
 /*
  * Copyright (c) 2011, Oracle and/or its affiliates. All rights reserved.
  * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
@@ -41,26 +43,49 @@
      * Seven-bit ASCII, a.k.a. ISO646-US, a.k.a. the Basic Latin block of the
      * Unicode character set
      */
-    public static final Charset US_ASCII = Charset.forName("US-ASCII");
+    public static final Charset US_ASCII = new US_ASCII();
     /**
      * ISO Latin Alphabet No. 1, a.k.a. ISO-LATIN-1
      */
-    public static final Charset ISO_8859_1 = Charset.forName("ISO-8859-1");
+    public static final Charset ISO_8859_1 = null;
     /**
      * Eight-bit UCS Transformation Format
      */
-    public static final Charset UTF_8 = Charset.forName("UTF-8");
+    public static final Charset UTF_8 = new UTF_8();
+    static final String[] aliases_UTF_8 = new String[] {
+            "UTF8",
+            "unicode-1-1-utf-8",
+        };
+
+        static final String[] aliases_ISO_8859_1 = new String[] {
+        "iso-ir-100",
+        "ISO_8859-1",
+        "latin1",
+        "l1",
+        "IBM819",
+        "cp819",
+        "csISOLatin1",
+        "819",
+        "IBM-819",
+        "ISO8859_1",
+        "ISO_8859-1:1987",
+        "ISO_8859_1",
+        "8859_1",
+        "ISO8859-1",
+    };
+
+
     /**
      * Sixteen-bit UCS Transformation Format, big-endian byte order
      */
-    public static final Charset UTF_16BE = Charset.forName("UTF-16BE");
+    public static final Charset UTF_16BE = null;
     /**
      * Sixteen-bit UCS Transformation Format, little-endian byte order
      */
-    public static final Charset UTF_16LE = Charset.forName("UTF-16LE");
+    public static final Charset UTF_16LE = null;
     /**
      * Sixteen-bit UCS Transformation Format, byte order identified by an
      * optional byte-order mark
      */
-    public static final Charset UTF_16 = Charset.forName("UTF-16");
+    public static final Charset UTF_16 = null;
 }
